tephra-hbase-compat-1.1-base/src/main/java/org/apache/tephra/hbase/SecondaryIndexTable.java

```diff
@@ -56,17 +58,24 @@
   private static final byte[] secondaryIndexQualifier = Bytes.toBytes('r');
   private static final byte[] DELIMITER  = new byte[] {0};
 
-  public SecondaryIndexTable(TransactionServiceClient transactionServiceClient, HTableInterface hTable,
-                             byte[] secondaryIndex) {
+  public SecondaryIndexTable(TransactionServiceClient transactionServiceClient, Table hTable,
+                             byte[] secondaryIndex) throws IOException {
     secondaryIndexTableName = TableName.valueOf(hTable.getName().getNameAsString() + ".idx");
-    HTable secondaryIndexHTable = null;
-    try (HBaseAdmin hBaseAdmin = new HBaseAdmin(hTable.getConfiguration())) {
+    Table secondaryIndexHTable = null;
+    Connection conn  = null;
+    try { 
+      conn  = ConnectionFactory.createConnection(hTable.getConfiguration());
+      Admin hBaseAdmin = conn.getAdmin();
       if (!hBaseAdmin.tableExists(secondaryIndexTableName)) {
         hBaseAdmin.createTable(new HTableDescriptor(secondaryIndexTableName));
       }
-      secondaryIndexHTable = new HTable(hTable.getConfiguration(), secondaryIndexTableName);
-    } catch (Exception e) {
+      secondaryIndexHTable = conn.getTable(secondaryIndexTableName);
+    } catch (Exception e) { 
       Throwables.propagate(e);
+    } finally {
+      if (conn != null) {
+         conn.close();
```

Comment:
This connection is local to the constructor and gets closed when the constructor execution is complete. Looks like we don't need to take any further actions in ``close()``. Please let me know otherwise.